### DISCUSSION TITLE
Combat, Earthquake spell functioning suggestions taking into account the hero's spell power

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#7493] [Ideas]
So, here are a few suggestions on the functioning of the Earthquake spell, taking into account the hero's spell power.

For castle defense, these values are currently used:
- 2 hit points per wall section
- 1 hit point for each turret
- 1 hit point for the drawbridge
- 1 additional hit point per wall section when fortifications are built in the Knight's castle
- 1 hit point for the Ballista

This corresponds to a total of 16 hit points for the Knight's castle when the fortifications are built, and a total of 12 hit points for the other castles.

Consider that the Ballista can also be hit by the Earthquake spell.

Let's keep the hit point values for the Earthquake spell the same as for the catapult, also with the constraint of not creating additional graphics for damage to castle walls, etc., and assume that each point of the hero's spell power does 0.5 damage (rounded down for these initial damage).
Let's also randomly add 0 or 1 damage to the result obtained.
Also, the initial damage cannot be less than 1 damage.

This gives us:
- spell power 1, 0.5 damage = 1 damage + (0 damage or 1 damage) --> 1 damage or 2 damage
- spell power 2, 1.0 damage = 1 damage + (0 damage or 1 damage) --> 1 damage or 2 damage
- spell power 3, 1.5 damage = 1 damage + (0 damage or 1 damage) --> 1 damage or 2 damage
- spell power 4, 2.0 damage = 2 damage + (0 damage or 1 damage) --> 2 damage or 3 damage
- spell power 5, 2.5 damage = 2 damage + (0 damage or 1 damage) --> 2 damage or 3 damage
- spell power 6, 3.0 damage = 3 damage + (0 damage or 1 damage) --> 3 damage or 4 damage
- spell power 7, 3.5 damage = 3 damage + (0 damage or 1 damage) --> 3 damage or 4 damage
- spell power 8, 4.0 damage = 4 damage + (0 damage or 1 damage) --> 4 damage or 5 damage
- spell power 9, 4.5 damage = 4 damage + (0 damage or 1 damage) --> 4 damage or 5 damage
- spell power 10, 5.0 damage = 5 damage + (0 damage or 1 damage) --> 5 damage or 6 damage
- spell power 11, 5.5 damage = 5 damage + (0 damage or 1 damage) --> 5 damage or 6 damage
- spell power 12, 6.0 damage = 6 damage + (0 damage or 1 damage) --> 6 damage or 7 damage
- spell power 13, 6.5 damage = 6 damage + (0 damage or 1 damage) --> 6 damage or 7 damage
- spell power 14, 7.0 damage = 7 damage + (0 damage or 1 damage) --> 7 damage or 8 damage
- spell power 15, 7.5 damage = 7 damage + (0 damage or 1 damage) --> 7 damage or 8 damage
- spell power 16, 8.0 damage = 8 damage + (0 damage or 1 damage) --> 8 damage or 9 damage
- spell power 17, 8.5 damage = 8 damage + (0 damage or 1 damage) --> 8 damage or 9 damage
- spell power 18, 9.0 damage = 9 damage + (0 damage or 1 damage) --> 9 damage or 10 damage
- spell power 19, 9.5 damage = 9 damage + (0 damage or 1 damage) --> 9 damage or 10 damage
- spell power 20, 10.0 damage = 10 damage + (0 damage or 1 damage) --> 10 damage or 11 damage
- spell power 21, 10.5 damage = 10 damage + (0 damage or 1 damage) --> 10 damage or 11 damage
- spell power 22, 11.0 damage = 11 damage + (0 damage or 1 damage) --> 11 damage or 12 damage
- spell power 23, 11.5 damage = 11 damage + (0 damage or 1 damage) --> 11 damage or 12 damage
- spell power 24, 12.0 damage = 12 damage + (0 damage or 1 damage) --> 12 damage or 13 damage
- spell power 25, 12.5 damage = 12 damage + (0 damage or 1 damage) --> 12 damage or 13 damage
- spell power 26, 13.0 damage = 13 damage + (0 damage or 1 damage) --> 13 damage or 14 damage
- spell power 27, 13.5 damage = 13 damage + (0 damage or 1 damage) --> 13 damage or 14 damage
- spell power 28, 14.0 damage = 14 damage + (0 damage or 1 damage) --> 14 damage or 15 damage
- spell power 29, 14.5 damage = 14 damage + (0 damage or 1 damage) --> 14 damage or 15 damage
- spell power 30, 15.0 damage = 15 damage + (0 damage or 1 damage) --> 15 damage or 16 damage
- spell power 31, 15.5 damage = 15 damage + (0 damage or 1 damage) --> 15 damage or 16 damage
- spell power 32+, 16.0 damage = 16 damage + (0 damage or 1 damage) --> 16 damage

So, an average hero with a spell power of 10 will need to cast the Earthquake spell two or three times to destroy all the castle's defenses (excluding the catapult shots).
A hero will need a spell power of 22 to possibly destroy all the castle's defenses with a single Earthquake spell cast.
A hero will need a spell power of 24+ to be guaranteed to destroy all the castle's defenses with a single Earthquake spell cast.

For the Knight's castle, if the fortifications are built, the hero must have a spell power of 30 to possibly destroy all the castle's defenses with a single Earthquake spell cast.
A hero will need a spell power of 32+ to be guaranteed to destroy all the Knight's castle defenses, with the fortifications built, with a single Earthquake spell cast.

Earthquake spell damage is randomly distributed across the city's defenses.
However, as with actions during combat, this is a determined randomness, with the same actions resulting in the same damage in the same places for the city's defenses.

As walls must be destroyed from the top and not the bottom, if the Knight's castle fortifications are built, they are hit first.

The Ballista is also considered stronger than the castle's other defenses.
It is consequently more likely not to be hit directly.

All parts therefore have the same chance of being hit, these are:
- each wall fortification, if present (1 hit point)
- each wall, two parts, starting with the top of the wall (2 hit points)
- each turret (1 hit point)
- drawbridge (1 hit point)

The Ballista (1 hit point) is three times less likely to be hit.

Damage is distributed over two phases.
One point of damage is subtracted from the total damage points.
The "reduced" damage is applied, then the substracted damage point is applied.

The aim is to make it possible to hit the same portion of wall twice with the same Earthquake spell cast, so that it is less predictable in the way it hits the castle's defenses (less boring).

Here are a few examples:
Earthquake spell does 4 damage:
- 1 point of damage is subtracted from total damage points
- 3 points of damage are applied to 3 high parts of 3 walls
- 1 point of damage subtracted is applied to 1 base part of one of the three walls previously hit

Earthquake spell does 4 damage:
- 1 point of damage is subtracted from total damage points
- 3 points of damage are applied to 3 high parts of 3 walls
- 1 point of damage subtracted is applied to 1 turret

Earthquake spell does 4 damage:
- 1 point of damage is subtracted from total damage points
- 2 points of damage are applied to 2 high parts of 2 walls
- 1 point of damage is applied to the drawbridge
- 1 point of damage subtracted is applied to 1 turret

**Comment by oleg-derevenetz** [comment]
Proposed mechanics is not quite clear to me. What if we have, say, 5-6 damage (hero with SP 10), but just two walls (suppose that other walls are already destroyed by the catapult). How exactly should this work? Should the Earthquake be guaranteed to destroy the remaining walls or not? What should be the mechanism for distributing excess damage, especially in cases when not all fortifications are built (e.g. castle has no Turrets), or some of them are already destroyed?

**Reply by LeHerosInconnu** [reply]
Hello @oleg-derevenetz,

> What if we have, say, 5-6 damage (hero with SP 10), but just two walls (suppose that other walls are already destroyed by the catapult). How exactly should this work?

If there is 6 damage and 8 hit points remain for the castle defenses, 6 hit points are destroyed, and 2 hit points remain for the castle defenses.

If there is 6 damage and 2 hit points remain for the castle defenses, 2 hit points are destroyed, and 0 hit point remain for the castle defenses (the remaining castle defenses are completely destroyed).

> Should the Earthquake be guaranteed to destroy the remaining walls or not?

Yes.
All damage is fully applied.

Think of it as the hit points of a creature and the damage of a magic damage spell like Lightning Bolt.
The hero has a spell power of 8.
The Lightning Bolt spell therefore does 8 x 25 damage, for a total of 200 damage.
The hero targets a titan with 300 hit points.
The Lightning Bolt spell does 200 damage, leaving the titan with 100 hit points.
The next turn, the hero targets the same titan with 100 hit points left.
The Lightning Bolt spell does 200 damage, more than enough to decimate the titan.
The titan is completely decimated.

> What should be the mechanism for distributing excess damage, especially in cases when not all fortifications are built (e.g. castle has no Turrets), or some of them are already destroyed?

Defenses present when the spell is cast are taken into account for damage distribution.
If there are more defenses present (more hit points) than damage to be applied, some defenses are not destroyed.
If there is more damage to be applied than defenses present (less hit points), all remaining defenses are completely destroyed.

**Reply by oleg-derevenetz** [reply]
It turns out to be an earthquake of selective action - strictly under intact fortifications.

**Reply by LeHerosInconnu** [reply]
Earthquake can't affect what no longer exists, and what remains is already fragile. :)

**Comment by Alucard648** [comment]
I think, for Earthquake spell, 1SP should count as 1 catapult shot at Basic Ballistics skill, or no skill, if testing results in spell being OP, despite huge mana cost.

**Reply by LeHerosInconnu** [reply]
This would make the Earthquake spell really too powerful.
The spell cast by a hero with a spell power of 10 would be like ten catapult shots, like 10 turns of catapult shots.

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 
That's well thought approach for sure. 👍 
But I'm still hesitating whether the spell should be affected by SP. In the original game this spell could be used on the same level by magic and might heroes.
With the proposed logic, magic heroes have a significant advantage, while might heroes won't use this spell at all due to low effectiveness. 
For example, in the OG Knight heroes have low SP, so during a siege they have quite a limited number of magic tricks. With the original logic of this spell Knight hero could use it and instantly break into a castle (considering his ballistics skill as well). Knight faction has no fliers and weak shooters, so using earthquake spell looks like a real tactics during siege. With the proposed logic it won't be the case and Knight heroes (who has really low level of SP) won't use it at all.
And on contrary: Magic heroes with high SP can break into castle really fast (with the proposed logic), while in the original game they had to spent more time breaking through castle walls or destroying turrets (which lead to more losses).

**Reply by LeHerosInconnu** [reply]
Thanks. :)

It's true that in the original game, a hero with a low level of spell power can open a breach in the castle walls on the first casting of the spell, but he can also fail to open a breach at all.
It's this unpredictable, unreliable side of the spell that generally makes that it's more interesting to cast another spell, and also which costs far less in spell points.
Fifteen spell points to cast a spell that is neither predictable nor reliable is a bit much for a hero of might.
A slow or haste spell is less costly and often allows rangers, for example, to come into play before opposing ranged or flying creatures.

In my opinion, it's better to have a reliable spell that does less damage than one that does more damage but the reality of that damage is uncertain.

It's also true that other spells work in a way that doesn't take into account the hero's spell power, such as teleport or dispel magic, but these spells are reliable in their effect and have a known and expected operation.
In my many games of Heroes 2, I used the Earthquake spell at the beginning, but hardly ever afterwards, because risking 15 spell points for a random result is a bit of a waste.
And the higher the hero's spell power, the less desirable the spell, as it's better, for example, to do 250 points of damage to an opposing troop with certainty than to maybe break a few pieces of wall.

**Comment by idshibanov** [comment]
Just for comparison sake, in OG earthquake has flat 75% chance to damage every wall section, 15% to deal double damage, 50% to break a tower and 25% for the moat bridge. Every roll is made independently and it adds up to 6 damage on average. It will require 12 spell power to achieve similar result with current suggested values.

I'm not against spellpower scaling, but we have to be careful not to weaken an already unpopular spell.

**Reply by LeHerosInconnu** [reply]
Hello @idshibanov,

Thank you for all these details.
And as you indicate, this spell is not very popular.
Here's my point of view on the subject:

...
The problem is that in the original game, the Earthquake spell is too random in its effects.
As a result, the spell is only really usable in very specific cases for the human player.
The AI uses it sometimes, and generally with significant results.

As far as I'm concerned, the Earthquake spell should have effects that are fully known when it's cast (in terms of amount of damage), and that also depend on the hero's spell power, as is the case with direct damage spells such as Lightning.
Frankly, in the original game, it's more interesting to cast another spell in the vast majority of cases.

I think the spell could work according to the hero's spell power (and therefore be predictable in its power, like damage spells) without it being too powerful for a magic hero.
I'll refine an idea I've already had on this subject, once I've had answers to my questions, earlier in this post, about hit point values. :)

_Originally posted by @LeHerosInconnu in https://github.com/ihhub/fheroes2/issues/7489#issuecomment-1656850847_
            
Related issue: https://github.com/ihhub/fheroes2/issues/7489.

**Comment by zenseii** [comment]
My suggestion is quite simple:
Keep the original functioning of this spell, and design a new spell from scratch that is better designed/more powerful. There can exist more than 1 spell that damages castle walls.

**Reply by LeHerosInconnu** [reply]
Hello @zenseii,

Making two redundant spells is not a solution.
And it's also going to "pollute" the mage guild with spells that are quite rarely used (only during a siege).

**Reply by zenseii** [reply]
Only during sieges is frankly quite often in Heroes 2. Also upgraded spells are a thing - Earthquake could easily be the crappy version. I wouldn't call "upgraded" and non-upgraded spells a redundancy of the game. If they function differently in a way that players understand it then it's fine.

**Reply by LeHerosInconnu** [reply]
Hello @zenseii,

My point is to compare the number of combats on the adventure map against wandering creatures and heroes with the number of combats during a siege.
The number of siege combats is much smaller.
Now the Earthquake spell is not necessarily useful, as catapult damage may be sufficient to avoid using the spell.

This already gives us a reduced use of the spell compared to other spells (over the course of a game).
And now we'd have to choose between two similar spells, one of which is random (the way the spell works in the original game) and the other which has known effects and is reliable.
This is a strategy game, so I prefer a reliable spell with known effects to one with random effects.
I don't see the point of using a random spell when a reliable one is available.

I'd also rather have other more useful spells in my expensive mage guilds than a similar second spell that's relatively less used.

**Reply by zenseii** [reply]
Hi, @LeHerosInconnu.


> My point is to compare the number of combats on the adventure map against wandering creatures and heroes with the number of combats during a siege. The number of siege combats is much smaller. Now the Earthquake spell is not necessarily useful, as catapult damage may be sufficient to avoid using the spell.

Virtually every map has a siege, certainly some sieges are just against basic towns. Meanwhile, not all maps have a sea (summon boat), not all have necromancers (holy word, holy shout, animate dead), and Dragon slayer is also an extremely specific spell, yes dragons are powerful but what if a map doesn't have a necromancer and warlock (dragon city is even more specific) it would basically depend on neutral armies either randomly having them or dragons being manually placed there.

> This already gives us a reduced use of the spell compared to other spells (over the course of a game). And now we'd have to choose between two similar spells, one of which is random (the way the spell works in the original game) and the other which has known effects and is reliable. ¨

> This is a strategy game, so I prefer a reliable spell with known effects to one with random effects. I don't see the point of using a random spell when a reliable one is available.

Obviously you will choose the better spell, IF you can afford to use the spell points. If your hero is too poor on spell points then the 30 sp cost of the better spell, as an example, might not be better than the one that costs 15 sp. Spell points are and should be a very limited resource in Heroes 2, and it is part of the gameplay to make careful considerations about spending them.

It is indeed a strategy game, which is why you need to make hard but fun decisions about risk/cost vs. potential benefit. The randomness of the earthquake spell is typical game design - it could work or it could not. This creates suspense, and excitement when it works, much like whether a chest contains 2000 gold or 1000 gold, or a good/crappy artifact. I'm sure you wouldn't change it so that every chest always contains the same predictable content?


> I'd also rather have other more useful spells in my expensive mage guilds than a similar second spell that's relatively less used.

This is a good point and I agree with it. Having 1 extra spell would decrease the chances of getting other spells, so we cannot just pollute the mage guild with lots of spells without carefully considering them.

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10012.
I close the discussion.
